# Measures of Dispersion

## Objectives ##
- Calculate the standard deviation of a data set.
- Distinguish between the population standard deviation and sample standard deviation.
- Identify the outliers of a data set. 

## The Standard Deviation ##
An important characteristic of any set of data is the variation in the data. In some data sets, the data values are concentrated closely near the mean; in other data sets, the data values are more widely spread out from the mean. The most common measure of variation is the standard deviation. The **standard deviation** is a number that measures the average distance of data values from their mean.

Because distance is never a negative value, the standard deviation is always positive or zero. The standard deviation is small when the data are all concentrated close to the mean, exhibiting little variation. The standard deviation is larger when the data values are more spread out from the mean, exhibiting more variation.

Suppose that we are studying the amount of time customers wait in line at the checkout at supermarket A and supermarket B. the average wait time at both supermarkets is five minutes. At supermarket A, the standard deviation for the wait time is two minutes; at supermarket B the standard deviation for the wait time is four minutes.

Because supermarket B has a higher standard deviation, we know that there is more variation in the wait times at supermarket B. Overall, wait times at supermarket B are more spread out from the mean; wait times at supermarket A are more concentrated near the mean.

In [18]:
png("sd_example.png", width = 1000, height = 1000)

library("beeswarm")

rangebrace = function(x, y0, y1, d, col){
    X = seq(x, x-d, length = 500)
    Ytop = (y1 - y0)/4 * ( (2/d)^(1/3) * sign(X - (x - d/2))*abs(X - (x - d/2))^(1/3) + 1 ) + (y0 + y1)/2
    Ybot = (y1 - y0)/4 * ( -(2/d)^(1/3) * sign(X - (x - d/2))*abs(X - (x - d/2))^(1/3) - 1 ) + (y0 + y1)/2
    lines(X, Ytop, lwd = 3, col = col)
    lines(X, Ybot, lwd = 3, col = col)
}

rangebar = function(x, y0, y1, d, col){
    segments(x-d, y0, x+d, y0, lwd = 3, col = col)
    segments(x, y0, x, y1, lwd = 3, col = col)
    segments(x-d, y1, x+d, y1, lwd = 3, col = col)
}

x1 = rnorm(100, mean = 3, sd = 1.5)
x2 = rnorm(100, mean = -1, sd = 3.5)

par(mar = c(2, 3.5, 2, 0))
beeswarm(x = list(x1, x2), pch = 21, col = "black", bg = "gray70", 
         cex = 2.5, cex.axis = 2, cex.main = 2, main = "The Standard Deviations of Two Data Sets", xlim = c(0.4, 2.4), xaxt = "n", yaxt = "n")
axis(1, at = c(1, 2), labels = c("Data Set 1", "Data Set 2"), cex.axis = 2)
axis(2, at = seq(floor(min(c(x1), c(x2))), ceiling(max(c(x1),c(x2))), 1), cex.axis = 2, las = 2)

d = 0.2

segments(1.0 - d, mean(x1), 1.0 + d, mean(x1), lwd = 4, lty = "dotted", col = "blue3")
segments(2.0 - d, mean(x2), 2.0 + d, mean(x2), lwd = 4, lty = "dotted", col = "blue3")

text(x = 1.0 + d, y = mean(x1), labels = "Mean\nData Set 1", cex = 2, col = "blue3", pos = 4)
text(x = 2.0 + d, y = mean(x2), labels = "Mean\nData Set 2", cex = 2, col = "blue3", pos = 4)

eps = 0.05
rangebar(x = 1.0 - d, y0 = mean(x1) + eps, y1 = mean(x1) + sd(x1), d = 0.02, col = "red3")
rangebar(x = 1.0 - d, y0 = mean(x1) - sd(x1), y1 = mean(x1) - eps, d = 0.02, col = "red3")
rangebar(x = 2.0 - d, y0 = mean(x2) + eps, y1 = mean(x2) + sd(x2), d = 0.02, col = "red3")
rangebar(x = 2.0 - d, y0 = mean(x2) - sd(x2), y1 = mean(x2) - eps, d = 0.02, col = "red3")

text(x = 1.0-d, y = mean(x1) + 0.5*sd(x1), labels = "Standard Deviation\nData Set 1", cex = 2, col = "red3", pos = 2)
text(x = 1.0-d, y = mean(x1) - 0.5*sd(x1), labels = "Standard Deviation\nData Set 1", cex = 2, col = "red3", pos = 2)
text(x = 2.0-d, y = mean(x2) - 0.5*sd(x2), labels = "Standard Deviation\nData Set 2", cex = 2, col = "red3", pos = 2)
text(x = 2.0-d, y = mean(x2) + 0.5*sd(x2), labels = "Standard Deviation\nData Set 2", cex = 2, col = "red3", pos = 2)


dev.off()

png 
  2

```{figure} sd_example.png
---
width: 100%
alt: Two data sets side-by-side. The first data set has data more squashed close to the mean and a small standard deviation. The second data set has data more spread out from the mean and a larger standard deviation.
name: sd-example
---
The standard deviation is a measure of the average distance of data values from their mean. If we compare the two data sets in the figure, the values of data set 1 are squashed closer to the mean, so data set 1 has a smaller standard deviation. The values of data set 2 are more spread out from the mean, so data set 2 has a larger standard deviation.

## Calculating the Population Standard Deviation

We will first look at calculating the standard deviation for a population. If $x$ is a data value, then the difference "$x – \text{mean}$" is called its **deviation**; it's a measure of the distance from the value $x$ to the mean of the data. There is a deviation for each value in a data set. The deviations are used to calculate the standard deviation. For a population, the deviation for a data value $x$ is $x – \mu$.

For example, suppose a population has mean $\mu = 5$, and consider two data values of the population: $x = 2$ and $x = 8$. The deviation of the value $x = 2$ from the mean is 

$$x - \mu = 2 - 5 = -3.$$ 

The deviation of the value $x = 8$ from the mean is 

$$x - \mu = 8 - 5 = 3.$$

Both $x = 2$ and $x = 8$ are a distance of $3$ units away from $\mu = 5$. But the signs (positives or negatives) of the deviations also tell us direction: since the deviation of $x = 2$ is $-3$, $x = 2$ is $3$ units *below* the mean; since the deviation of $x = 8$ is $+3$, $x = 8$ is $3$ units *above* the mean.

When calculating the standard deviation, we want to only focus on the average distance from the mean, not the direction. To remove the information on direction (the sign) from each deviation, we square each deviation. For $x = 2$, the squared deviation is

$$ (x - \mu)^2 = (2 - 5)^2 = (-3)^2 = 9. $$

For $x = 8$, the squared deviation is

$$ (x - \mu)^2 = (8 - 5)^2 = 3^2 = 9. $$

Since squaring a deviation always results in a positive number, it removes the information on direction. Unfortunately, squared deviations tell us the *square* of the distance of each value from the mean, not simply the distance. We will correct this problem later.

Next, we find the mean of all the squared deviations in the usual way: by adding up all the squared deviations in the population and dividing by the population size $N$:

$$ \frac{\sum(x - \mu)^2}{N}. $$

This is the average *squared* distance of the population data from the mean. We need to "unsquare" the average by taking the square root. This gives us the formula for the population standard deviation:

$$ \sigma = \sqrt{\frac{\sum(x - \mu)^2}{N}}.$$

We denote the population standard deviation by $\sigma$ (the lower-case Greek letter sigma).

Note that to calculate the population standard deviation $\sigma$, we first must calculate the population mean $\mu$ since we use $\mu$ in the formula for $\sigma$. 

Also, to use the formula for population standard deviation in R, we will need to know how to take a square root in R. We can take the square root of a value in R using the <code>sqrt</code> function:

```R
sqrt(x)
```

where <code>x</code> is a number or list of numbers. For example,

In [19]:
sqrt(4)

[1] 2

***

### Example 2.3.1

The Cook Political Report scores the partisan lean of each state in the U.S. based on the results of recent elections. The Partisan Voting Index (PVI) for each of the 50 states (as of April 2021) is below, where more negative values means the state leans more toward the Democratic party, and more positive values means the state leans more toward the Republican party.[^attribution]

<center>
    15, 9, 3, 16, -14, -3, -7, -6, 3, 3, -15, 19, -7, 11, 6, 11, 16, 12, -1, -14, -14, 1, -1, 10, 11, 11, 13, 0, 0, -6, -3, -10, 3, 20, 6, 20, -6, 2, -8, 8, 16, 14, 5, 13, -15, -2, -8, 23, 2, 26
</center>
<br/>

1. Calculate the mean state PVI.
2. Calculate the standard deviation of the state PVIs.

[^attribution]: Wasserman, David; Flinn, Ally (April 15, 2021). "[Introducing the 2021 Cook Political Report Partisan Voter Index](https://cookpolitical.com/analysis/national/pvi/introducing-2021-cook-political-report-partisan-voter-index)". The Cook Political Report. Retrieved April 15, 2021.


#### Solution
Note that this list includes the PVIs for the entire *population* of 50 states, so we will be calculating the *population* mean and the *population* standard deviation.

##### Part 1

In [22]:
x = c(15, 9, 3, 16, -14, -3, -7, -6, 3, 3, -15, 19, -7, 11, 6, 11, 16, 12, -1, -14, -14, 1, -1, 10, 11, 11, 13, 0, 0, -6, -3, -10, 3, 20, 6, 20, -6, 2, -8, 8, 16, 14, 5, 13, -15, -2, -8, 23, 2, 26)

# Calculate the Mean
N = length(x)

mu = sum(x)/N
mu

[1] 3.76

The mean PVI is $\mu = 3.76$. This means that states on average lean Republican.

##### Part 2

In [23]:
# Calculate the Standard Deviation
sigma = sqrt( sum( (x - mu)^2 )/N )
sigma

[1] 10.67438

The standard deviation of the state PVIs is $\sigma = 10.6744$. On average, state PVIs differ from the mean by $10.6744$ points.

```{warning}
Be careful that you don't interpret the mean state PVI that we calculated above to be the same as the PVI for the whole United States. Many Democratic leaning states have much larger populations than Republican leaning states, but we didn't account for population differences in our analysis. For example, California (with a PVI of -14) has a population that is 68.5 times as large as the population of Wyoming (with a PVI of 26), but both states were treated equally in our calculations above. The lesson: we must be very careful not to misinterpret or over-interpret statistical results.
```

***

## Calculating the Sample Standard Deviation

In many real-life situations, it is impractical to collect the data needed to calculate the standard deviation for an entire population. Instead, statisticians estimate the standard deviation of a population by calculating the standard deviation of a sample. The formula for the sample standard deviation is almost the same as the formula for the population standard deviation.

**Formula for the Sample Standard Deviation:**

$$ s = \sqrt{\frac{\sum(x - \bar{x})^2}{n - 1}} $$

Here, $x$ is a data value in our sample, $\bar{x}$ is the sample mean, and $n$ is the sample size. We denote the sample standard deviation by $s$.

For comparison, let's also review the formula for calculating the population standard deviation.

**Formula for the Population Standard Deviation:**

$$ \sigma = \sqrt{\frac{\sum(x - \mu)^2}{N}} $$

Here, $x$ is a data value in our population, $\mu$ is the population mean, and $N$ is the population size. We denote the population standard deviation by $\sigma$.

Most of the differences between the sample standard deviation $s$ and the population standard deviation $\sigma$ are trivial. Because we are dealing with a sample instead of a population, we use $\bar{x}$ instead of $\mu$ to notate the mean, and we use $n$ instead of $N$ to notate the number of data values. But the most significant difference is that the when calculating the sample standard deviation, we divide by *one less* than the number of data values: $n - 1$. In contrast, when calculating the population standard deviation, we simply divide by the number of data values: $N$.

```{admonition} Why?
:class: tip
Why do we divide by $n - 1$ instead of $n$ in the sample standard deviation formula? The sample standard deviation uses the distances from each value to the *sample* mean in its calculation: $(x - \bar{x})$. We want the sample standard deviation to approximate the population standard deviation, but the population standard deviation averages the distances from each value to the *population* mean: $(x - \mu)$. The sample mean $\bar{x}$ is usually close to $\mu$, but they are rarely the same. Because of this small difference in what is being calculated, we need a small correction to the formula. Using techniques that are more advanced than what we will cover in this text, statisticians can show that dividing by $n - 1$ instead of $n$ corrects this bias.
```

***


### Example 2.3.2 ###
Use data below represents students' scores on the first exam from one of Susan Dean's precalculus classes.

33, 42, 49, 49, 53, 55, 55, 61, 63, 67, 68, 68, 69, 69, 72, 73, 74, 78, 80, 83, 88, 88, 88, 90, 92, 94, 94, 94, 94, 96, 100

1. Calculate the sample mean.
2. Calculate the sample standard deviation.

#### Solution ####
##### Part 1 #####

In [3]:
x = c(33, 42, 49, 49, 53, 55, 55, 61, 63, 67, 68, 68, 69, 69, 72, 73, 74, 78, 80, 83, 88, 88, 88, 90, 92, 94, 94, 94, 94, 96, 100)

# Calculate the Sample Mean
n = length(x)

xbar = sum(x)/n
xbar

[1] 73.51613

The mean score is $\bar{x} = 73.5161$.

##### Part 2 #####

In [4]:
# Calculate the Sample Standard Deviation
s = sqrt( sum( (x - xbar)^2 ) / (n - 1) )
s

[1] 17.92367

So the standard deviation of the scores is $s = 17.9237$.

***

## Outliers

**Outliers** are data values that deviate from the mean much greater than the standard amount for the data. Statisticians are interested in identifying outliers because they may indicate an error in the data. For example, suppose we collect the following data for the price of a cup of coffee at several local coffee shops:

<center>
    2.69, 2.79, 2.99, 3.00, 2.69, 3.15, 2.89, 2.69, 2.90, <b>259</b>, 2.79, 2.80, 3.10, 2.59, 2.75
</center>
<br/>

The value in bold is a clear outlier. It seems extremely unlikely that this cup of coffee actually cost \$259. Instead, the data value may have been recorded incorrectly, so that the true cost for this cup of coffee was \$2.59, not \$259.

There are many different methods statisticians use to calculate outliers depending on the distribution of the data they are working with. We will learn one simple rule often used for detecting outliers. We will consider a data value an outlier if it is more than two standard deviations away from the mean. In mathematical notation, this means a data value is an outlier in a population if it is more than $\mu + 2\sigma$ or less than $\mu - 2\sigma$. In a sample, a data value is an outlier if it is more than $\bar{x} + 2s$ or less than $\bar{x} - 2s$. This is a reasonable definition because the average distance between data values and their mean is one standard deviation, so most data values lie within a distance of two standard deviations from their mean. Only relatively extreme data values are more than two standard deviations away from the mean.

This definition is more a "rule of thumb" than a rigid law. Professional statisticians may use different rules to identify outliers depending on the circumstances.

Also, just because a value is an outlier does not mean there is anything wrong with it. While some outliers may be the result of errors, other outliers may be perfectly valid data values that are just unusually large or small. You should determine whether outliers are valid on a case-by-case basis.

In [41]:
png("outlier_example.png", width = 1000, height = 1000)

library("beeswarm")
library("shape")

rangebrace = function(x, y0, y1, d, col){
    X = seq(x, x+d, length = 500)
    Ytop = (y1 - y0)/4 * ( -(2/d)^(1/3) * sign(X - (x + d/2))*abs(X - (x + d/2))^(1/3) + 1 ) + (y0 + y1)/2
    Ybot = (y1 - y0)/4 * ( (2/d)^(1/3) * sign(X - (x + d/2))*abs(X - (x + d/2))^(1/3) - 1 ) + (y0 + y1)/2
    lines(X, Ytop, lwd = 3, col = col)
    lines(X, Ybot, lwd = 3, col = col)
}

rangebar = function(x, y0, y1, d, col){
    segments(x-d, y0, x+d, y0, lwd = 3, col = col)
    segments(x, y0, x, y1, lwd = 3, col = col)
    segments(x-d, y1, x+d, y1, lwd = 3, col = col)
}

x = rnorm(100, sd = 1.5)

par(mar = c(2, 3.5, 2, 0))
beeswarm(x, pch = 21, col = "black", bg = "gray70", 
         cex = 2.5, cex.axis = 2, cex.main = 2, main = "Outliers", xaxt = "n", yaxt = "n", xlim = c(0.75, 1.5))
axis(2, at = seq(floor(min(x)), ceiling(max(x)), 1), cex.axis = 2, las = 2)

d = 0.1

segments(1.0 - d, mean(x) + 2*sd(x), 1.0 + d, mean(x) + 2*sd(x), lwd = 4, lty = "dotted", col = "blue3")
segments(1.0 - d, mean(x) + sd(x), 1.0 + d, mean(x) + sd(x), lwd = 4, lty = "dotted", col = "blue3")
segments(1.0 - d, mean(x), 1.0 + d, mean(x), lwd = 4, lty = "dotted", col = "blue3")
segments(1.0 - d, mean(x) - sd(x), 1.0 + d, mean(x) - sd(x), lwd = 4, lty = "dotted", col = "blue3")
segments(1.0 - d, mean(x) - 2*sd(x), 1.0 + d, mean(x) - 2*sd(x), lwd = 4, lty = "dotted", col = "blue3")

text(x = 1.0 - d, y = mean(x) + 2*sd(x), labels = "μ + 2σ", cex = 2, col = "blue3", pos = 2)
text(x = 1.0 - d, y = mean(x) + sd(x), labels = "μ + σ", cex = 2, col = "blue3", pos = 2)
text(x = 1.0 - d, y = mean(x), labels = "μ", cex = 2, col = "blue3", pos = 2)
text(x = 1.0 - d, y = mean(x) - sd(x), labels = "μ - σ", cex = 2, col = "blue3", pos = 2)
text(x = 1.0 - d, y = mean(x) - 2*sd(x), labels = "μ - 2σ", cex = 2, col = "blue3", pos = 2)

eps = 0.02
rangebar(1.0 + d, mean(x) + sd(x) + eps, mean(x) + 2*sd(x) - eps, d = 0.01, col = "red3")
rangebar(1.0 + d, mean(x) + eps, mean(x) + sd(x) - eps, d = 0.01, col = "red3")
rangebar(1.0 + d, mean(x) - eps, mean(x) - sd(x) + eps, d = 0.01, col = "red3")
rangebar(1.0 + d, mean(x) - sd(x) - eps, mean(x) - 2*sd(x) + eps, d = 0.01, col = "red3")

text(x = 1.0 + d, y = mean(x) + 3/2*sd(x), labels = "σ", cex = 2, col = "red3", pos = 4)
text(x = 1.0 + d, y = mean(x) + 1/2*sd(x), labels = "σ", cex = 2, col = "red3", pos = 4)
text(x = 1.0 + d, y = mean(x) - 1/2*sd(x), labels = "σ", cex = 2, col = "red3", pos = 4)
text(x = 1.0 + d, y = mean(x) - 3/2*sd(x), labels = "σ", cex = 2, col = "red3", pos = 4)

rangebrace(1.0 + 1.5*d, mean(x) + eps, mean(x) + 2*sd(x) - eps, d = 0.05, col = "red3")
rangebrace(1.0 + 1.5*d, mean(x) - eps, mean(x) - 2*sd(x) + eps, d = 0.05, col = "red3")

text(x = 1.0 + 1.5*d + 0.05, y = mean(x) + sd(x), labels = "Data values within\ntwo standard deviations\nabove the mean", cex = 2, col = "red3", pos = 4)
text(x = 1.0 + 1.5*d + 0.05, y = mean(x) - sd(x), labels = "Data values within\ntwo standard deviations\nbelow the mean", cex = 2, col = "red3", pos = 4)

delta = 0.02

start = 1/3*(mean(x) + 2*sd(x)) + 2/3*max(x)
outliers_above = x[x > mean(x) + 2*sd(x)]
for (outlier in outliers_above) {
    Arrows(1.2, start, 1 + delta, outlier, lwd = 3, arr.type = "triangle", arr.width = 0.3, col = "forestgreen")
}
if (length(outliers_above) > 0) {
    text(x = 1.2, y = start, labels = ifelse(length(outliers_above) > 1, "Outliers", "Outlier"), cex = 2, col = "forestgreen", pos = 4)
}

start = 1/3*(mean(x) - 2*sd(x)) + 2/3*min(x)
outliers_below = x[x < mean(x) - 2*sd(x)]
for (outlier in outliers_below) {
    Arrows(1.2, start, 1 + delta, outlier, lwd = 3, arr.type = "triangle", arr.width = 0.3, col = "forestgreen")
}
if (length(outliers_below) > 0) {
    text(x = 1.2, y = start, labels = ifelse(length(outliers_below) > 1, "Outliers", "Outlier"), cex = 2, col = "forestgreen", pos = 4)
}

dev.off()

png 
  2

```{figure} outlier_example.png
---
width: 100%
alt: The values of a data set are shown, and the outliers of the data set are indicated.
name: outlier-example
---
Given a set of data, most data values are within two standard deviations from their mean, either above or below the mean. In this text, we consider outliers to be data values that are more than two standard deviations away from the mean.
```

***

### Example 2.3.3

A dating website surveyed 50 of their recently-married customers to see how many long-term relationships they had before they were married. The results are below.

<center>
    1, 1, 1, 2, 2, 2, 2, 2, 2, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 4, 4, 4, 4, 4, 4, 5, 5, 5, 5, 5, 6, 6, 6, 7, 7, 7, 7, 8, 8, 8, 8, 9, 10, 11, 11, 13, 13, 14, 15
</center>
<br/>

1. Find the mean number of relationships of the individuals surveyed.
2. Find the standard deviation of the data.
3. Does the data contain any outliers? If so, what are they?

#### Solution
Note that the 50 customers surveyed are only a small *sample* of all the dating website's customers, so we will calculate the *sample* mean and *sample* standard deviation.

##### Part 1

In [31]:
x = c(1, 1, 1, 2, 2, 2, 2, 2, 2, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 4, 4, 4, 4, 4, 4, 5, 5, 5, 5, 5, 6, 6, 6, 7, 7, 7, 7, 8, 8, 8, 8, 9, 10, 11, 11, 13, 13, 14, 15)

# Calculate the Mean
n = length(x)

xbar = sum(x)/n
xbar

[1] 5.42

On average, those surveyed had $\bar{x} = 5.42$ relationships before they were married.

##### Part 2

In [32]:
s = sqrt( sum( (x - xbar)^2 )/(n - 1) )
s

[1] 3.55763

The sample standard deviation is $s = 3.558$ relationships.

##### Part 3
To find any outliers, we will calculate two standard deviations above the mean ($\bar{x} + 2s$) and two standard deviations below the mean ($\bar{x} - 2s$). Any data values above $\bar{x} + 2s$ or below $\bar{x} - 2s$ are outliers.

In [33]:
xbar + 2*s
xbar - 2*s

[1] 12.53526

[1] -1.695261

Returning to the original list of data values, we can see that $13$, $13$, $14$, and $15$ are greater than $\bar{x} + 2s = 12.5353$. Any values smaller than $\bar{x} - 2s = -1.6953$ would also be outliers, but there aren't any values this small in the data.

So our data contains a total of four outliers: $13$, $13$, $14$, and $15$. 